### PR TITLE
Add docs about architecture (fixes #430)

### LINKED ISSUE #430: Internal architecture of kinto in documentation
1. I think there must be a document explaining internal architecture of kinto. What are core packages like pyramid, zope, cliquet and sqlalchemy (ORM ) and learning them will surely help new contributors. 

2. Could we also add a flow/sequence diagram explaining internal architecture. ?

### PR DESCRIPTION
**Pull request by leplatrem**
capture d ecran de 2016-02-29 17-27-27

### CODE REVIEW

**Comment by leplatrem** [comment]
@magopian your super powers of newcomer could be required here ;)

File: docs/concepts.rst [hunk 1]
```diff
@@ -103,3 +103,46 @@ See also
...
 * :blog:`Our blog post <en/handling-permissions>` on permissions design in *Kinto*
+
+
+.. _technical-architecture:
+
+Technical Architecture
+======================
+
+Every concept of bucket, group, collection or record leverages the REST *resource*
+notion of *Cliquet*.
```
Comment (magopian) [inline docs/concepts.rst]: Add a link to cliquet?

File: docs/concepts.rst [hunk 2]
```diff
@@ -103,3 +103,46 @@ See also
...
+
+*Cliquet* is a toolkit which brings abstractions and
+good practices to build HTTP microservices, such as data-driven REST APIs.
+
+*Cliquet* is relies on *Cornice* and *Pyramid* to implement the endpoints
```
Comment (magopian) [inline docs/concepts.rst]: typo: no "is"

Comment (magopian) [inline docs/concepts.rst]: typo: no "is"

Comment (magopian) [inline docs/concepts.rst]: Add links to cornice and pyramid?

File: docs/concepts.rst [hunk 3]
```diff
@@ -103,3 +103,46 @@ See also
...
+and HTTP request/response handling.
+
+.. image:: images/architecture.svg
```
Comment (magopian) [inline docs/concepts.rst]: Do you plan on adding "comments" like you did below for the sequence diagrams? Btw, are those "comments" visible somewhere in the final page (in the "alt" of the image?)

**Comment by magopian** [comment]
I like this, especially the sequence diagram which is pretty clear. The architecture diagram also kind of gives some general information, and "tags" (? the bullet points on the right) about what each "layer" does.
The architecture diagram is a good compromise I guess between something that is general enough (you don't want a detailed explanation of cornice here), but still gives pointers on what each layer is responsible for.

r+wc from me, thanks for this work!

**Comment by leplatrem** [comment]
Thanks for your positive feedback! I'll take your comments into account.

File: docs/concepts.rst (hunk 3 above)
Comment (leplatrem) [inline docs/concepts.rst]: The comments for the sequence diagram are only visible in the source. It is just to be able to edit the sequence diagram on the website. For architecture, it's an SVG so we can edit it directly in Inkscape

Comment (magopian) [inline docs/concepts.rst]: ah, of course, I understand now, thanks ;)

**Pull request merged**